default/python/AI/turn_state.py

```diff
@@ -18,9 +24,60 @@ def __init__(self):
         self.__have_computronium = False
         self.__best_pilot_rating = 1e-8
         self.__medium_pilot_rating = 1e-8
+        self.__planets = {}
 
-    def cleanup(self):
+    def update(self):
         self.__init__()
+        universe = fo.getUniverse()
+        for pid in universe.planetIDs:
+            planet = universe.getPlanet(pid)
+            self.__planets[pid] = PlanetInfo(pid, planet.speciesName, planet.owner, planet.systemID)
+
+    def get_empire_species_systems(self):
+        """
+        Return dict from system id to planet ids of empire with species.
+
+        :rtype: dict[int, list[int]]
+        """
+        # TODO: as currently used, is duplicative with combo of foAI.foAIstate.popCtrSystemIDs and foAI.foAIstate.colonizedSystems
+        empire_id = fo.empireID()
+        planets_with_species = (x for x in self.__planets.itervalues() if x.owner == empire_id and x.species)
+        result = {}
+        for x in planets_with_species:
+            result.setdefault(x.system_id, []).append(x.pid)
+        return result
```

Comment:
I'm trying to figure out your design concept and intentions for this module. 

Is there a reason why this (and the following) is an explicit public function instead of calling the code once in ```update()``` and implementing the dict as class property?